### DISCUSSION TITLE
Pending change: rename go package to github.com/k3s-io/k3s

### DISCUSSION THREAD

**Discussion by luthermonson** [k3s-io/k3s#4799] [General]
I'd like to propose cleaning up the repo a little bit and moving the official package name of the k3s project from `github.com/rancher/k3s` to `github.com/k3s-io/k3s`. This would be a change here: https://github.com/k3s-io/k3s/blob/master/go.mod#L1 and find/replacing all the imports throughout the project.

The impact on anyone importing k3s into other projects is that they would need to rename and/or do a replace in their go mod if they want updates after this change is implemented.

Here is the link to the gh generated dependents of the github.com/rancher/k3s project https://github.com/k3s-io/k3s/network/dependents?package_id=UGFja2FnZS0yMjY0OTI2MDgz and the list is basically rancher repos or rancher forks which we have control over to update.

**Comment by Oats87** [comment]
SGTM; we should probably do this along with a minor release bump (1.24?)

**Reply by briandowns** [reply]
Would there be any drawback to doing this during a patch release?

**Reply by Oats87** [reply]
I don't think so, other than potential frustration caused from a `go get` for any downstream consumers. 

If we're OK with that, then we should probably backport this across all branches we still maintain IMO (all or nothing)

**Reply by luthermonson** [reply]
look at the list, the downstream consumers are basically rancher forks and a couple k3s wrapper projects. i i dont think we're gonna hurt anyone's feelings.

i think we should backport too, i have no problems with that

**Comment by luthermonson** [comment]
https://github.com/k3s-io/k3s/pull/4846

**Comment by dereknola** [comment]
As of June 27, 2022, all active release branches have changed over to the new packaging name: master, release-1.23, release-1.22. I am closing this discussion chain as a result.
